bluesky/protocols.py

```diff
@@ -31,12 +53,39 @@ def success(self) -> bool:
 
 
 @runtime_checkable
-class Readable(Protocol):
+class Named(Protocol):
+    @property
+    def name(self) -> str:
```

Comment:
In our split of Device and Ability, do the following need a name, and if so what should they be:
- A Device that represents the areaDetector driver and plugins of a Pilatus detector
- An Ability that lets the Pilatus be software triggered
- An Ability that lets the Pilatus be flyscanned
- A Signal for the max intensity of a frame as calculated by the NDStats plugin

Reply:
Case 2 and 3 are the same device, but 2 abilities with the same name

Reply:
Case 4: Add `__getattr__` on `Readable` that wraps signals in readable interface, and top level Ability knows about hints etc. of its children